torch/testing/_internal/common_utils.py

```diff
@@ -789,27 +790,80 @@ def safeCoalesce(self, t):
 
         return tg
 
-    def assertEqual(self, x, y, prec=None, message='', allow_inf=False, exact_dtype=None):
+    # Some analysis of tolerance by logging tests from test_torch.py can be found
+    # in https://github.com/pytorch/pytorch/pull/32538.
+    # dtype name : (rtol, atol)
+    dtype_precisions = {
+        'float16': (0.001, 1e-5),
+        'bfloat16': (0.016, 1e-5),
+        'float32': (1.3e-6, 1e-5),
+        'float64': (1e-7, 1e-7),
+        'complex32': (1.3e-6, 1e-5),
+        'complex64': (1e-7, 1e-7),
+    }
+
+    # todo: implement numpy-like issubdtype
+    def is_integral(self, dtype):
+        dtypes = get_all_dtypes()
+        return dtype in dtypes and not dtype.is_floating_point
```

Comment:
The logic here... seems a bit odd? Why would we care if `dtype in dtypes`?

Reply:
this is just a hack until we have an issubdtype api, get_all_dtypes doens't have certain dtypes like quantized or complex that aren't integral.